File: app/src/main/java/huce/fit/appreadstories/account/update/AccountUpdateActivity.java
package huce.fit.appreadstories.account.update;

import android.os.Bundle;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ImageView;
import android.widget.Toast;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import huce.fit.appreadstories.R;

public class AccountUpdateActivity extends AppCompatActivity implements AccountUpdateView {

    private AccountUpdatePresenter mAccountUpdatePresenter;
    private ImageView ivBack;
    private EditText etName, etEmail, etOldPassword, etNewPassword, etBirthday;
    private ImageView ivDate;
    private Button btSave;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_account_update);

        init();
        processEvents();
    }

    private void init() {
        etName = findViewById(R.id.etName);
        etEmail = findViewById(R.id.etEmail);
        etOldPassword = findViewById(R.id.etPassword1);
        etNewPassword = findViewById(R.id.etPassword2);
        etBirthday = findViewById(R.id.etBirthday);
        ivBack = findViewById(R.id.ivBack);
        ivDate = findViewById(R.id.ivDate);
        btSave = findViewById(R.id.btSave);

        mAccountUpdatePresenter = new AccountUpdateImpl(this);
    }

    @Override
    public void getInfoAccount(String name, String email, String birthday) {
        etName.setText(name);
        etEmail.setText(email);
        etBirthday.setText(birthday);
    }

    private void processEvents() {
        ivBack.setOnClickListener(v -> finish());
        ivDate.setOnClickListener(v -> mAccountUpdatePresenter.openDatePicker());
        btSave.setOnClickListener(v -> {
            String name = etName.getText().toString().trim();
            String email = etEmail.getText().toString().trim();
            String oldPass = etOldPassword.getText().toString().trim();
            String newPass = etNewPassword.getText().toString().trim();
            String birthday = etBirthday.getText().toString().trim();

            if (name.equals("") || email.equals("") || birthday.equals("")) {
                Toast.makeText(AccountUpdateActivity.this, "Vui lòng nhập đầy đủ tên hiển thị!", Toast.LENGTH_SHORT).show();
            } else {
                int idAccount = mAccountUpdatePresenter.getSharedPreferences().getIdAccount();
                if (oldPass.equals("")) {
                    if (newPass.equals("")) {
                        mAccountUpdatePresenter.updateAccount(idAccount, email, name, birthday);
                    } else {
                        Toast.makeText(AccountUpdateActivity.this, "Vui lòng nhập mật khẩu hiện tại", Toast.LENGTH_SHORT).show();
                    }
                } else {
                    if (newPass.equals("")) {
                        Toast.makeText(AccountUpdateActivity.this, "Vui lòng nhập mật khẩu mới", Toast.LENGTH_SHORT).show();
                    } else {
                        mAccountUpdatePresenter.changePassword(idAccount, oldPass, newPass);
                    }
                }
            }
        });
    }

    @Override
    public void changeDatePicker(String dateStr) {
        etBirthday.setText(dateStr);
    }

    @Override
    public void update(int status) {
        switch (status) {
            case 0:
                Toast.makeText(AccountUpdateActivity.this, "Cập nhật tài khoản thất bại", Toast.LENGTH_SHORT).show();
                break;
            case 1:
                Toast.makeText(AccountUpdateActivity.this, "Cập nhật tài khoản thành công", Toast.LENGTH_SHORT).show();
                finish();
                break;
            case 2:
                Toast.makeText(AccountUpdateActivity.this, "Vui lòng kết nối mạng", Toast.LENGTH_SHORT).show();
                break;
            case 3:
                Toast.makeText(AccountUpdateActivity.this, "Định dạng ngày sinh không đúng", Toast.LENGTH_SHORT).show();
                break;
            case 4:
                Toast.makeText(AccountUpdateActivity.this, "Mật khẩu mới trùng với mật khẩu cũ", Toast.LENGTH_SHORT).show();
                break;
            case 5:
                Toast.makeText(AccountUpdateActivity.this, "Cập nhật mật khẩu thất bại", Toast.LENGTH_SHORT).show();
                break;
            case 6:
                Toast.makeText(AccountUpdateActivity.this, "Mật khẩu hiện tại không chính xác", Toast.LENGTH_SHORT).show();
                break;
        }
    }
}

File: app/src/main/java/huce/fit/appreadstories/account/update/AccountUpdateActivity.kt
package huce.fit.appreadstories.account.update

import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.ImageView
import androidx.appcompat.app.AppCompatActivity
import huce.fit.appreadstories.R
import huce.fit.appreadstories.util.AppUtil

class AccountUpdateActivity : AppCompatActivity(), AccountUpdateView {

    private lateinit var ivBack: ImageView
    private lateinit var etName: EditText
    private lateinit var etEmail: EditText
    private lateinit var etOldPassword: EditText
    private lateinit var etNewPassword: EditText
    private lateinit var etBirthday: EditText
    private lateinit var ivDate: ImageView
    private lateinit var btSave: Button
    private lateinit var accountUpdatePresenter: AccountUpdateImpl

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_account_update)
        init()
        processEvents()
    }

    private fun init() {
        etName = findViewById(R.id.etName)
        etEmail = findViewById(R.id.etEmail)
        etOldPassword = findViewById(R.id.etPassword1)
        etNewPassword = findViewById(R.id.etPassword2)
        etBirthday = findViewById(R.id.etBirthday)
        ivBack = findViewById(R.id.ivBack)
        ivDate = findViewById(R.id.ivDate)
        btSave = findViewById(R.id.btSave)
        accountUpdatePresenter = AccountUpdateImpl(this)
    }

    override fun getInfoAccount(name: String, email: String, birthday: String) {
        etName.setText(name)
        etEmail.setText(email)
        etBirthday.setText(birthday)
    }

    private fun processEvents() {
        ivBack.setOnClickListener { finish() }
        ivDate.setOnClickListener { accountUpdatePresenter.openDatePicker() }
        btSave.setOnClickListener {
            val name = etName.getText().toString().trim { it <= ' ' }
            val email: String = etEmail.getText().toString().trim { it <= ' ' }
            val oldPassword: String = etOldPassword.getText().toString().trim { it <= ' ' }
            val newPassword: String = etNewPassword.getText().toString().trim { it <= ' ' }
            val birthday: String = etBirthday.getText().toString().trim { it <= ' ' }
            if (name.isEmpty() || email.isEmpty() || birthday.isEmpty()) {
                AppUtil.setToast(this, "Vui lòng nhập đầy đủ tên hiển thị!")
            } else {
                val accountId = accountUpdatePresenter.getAccount().getAccountId()
                if (oldPassword.isEmpty()) {
                    if (newPassword.isEmpty()) {
                        accountUpdatePresenter.updateAccount(accountId, email, name, birthday)
                    } else {
                        AppUtil.setToast(this, "Vui lòng nhập mật khẩu hiện tại")
                    }
                } else {
                    if (newPassword.isEmpty()) {
                        AppUtil.setToast(this, "Vui lòng nhập mật khẩu mới")
                    } else {
                        accountUpdatePresenter.changePassword(accountId, oldPassword, newPassword)
                    }
                }
            }
        }
    }

    override fun changeDatePicker(dateStr: String) {
        etBirthday.setText(dateStr)
    }

    override fun update(status: Int) {
        when (status) {
            0 -> setToast("Cập nhật tài khoản thất bại")

            1 -> {
                setToast("Cập nhật tài khoản thành công")
                finish()
            }

            2 -> setToast("Vui lòng kết nối mạng")

            3 -> setToast("Định dạng ngày sinh không đúng")

            4 -> setToast("Mật khẩu mới trùng với mật khẩu cũ")

            5 -> setToast("Cập nhật mật khẩu thất bại")

            6 -> setToast("Mật khẩu hiện tại không chính xác")
        }
    }

    private fun setToast(content: String) {
        AppUtil.setToast(this, content)
    }
}
